### PR TITLE
HTTP/3 interop fixes

### PR DESCRIPTION
**Pull request by scalablecory**
This fixes issues identified against public HTTP/3 interop test servers:
- SETTINGS frame not being sent correct; some servers block sending a response until receiving SETTINGS.
- SETTINGS frame not being parsed correctly for servers with more than one SETTINGS value. (Only a single setting is currently defined, but servers are allowed to send more and clients are supposed to ignore them if they aren't understood -- Chromium server is sending a lot of unknown settings)
- Possible double-free (back to ArrayPool.Default) if server control stream parsing fails.
- Remove some arbitrary frame size limits; it's not clear what values are reasonable here and it hurts interop -- will revisit these for .NET 6 once we have more solid implementations to look at.

Additionally, add/clarify some comments.

(recommend a first-pas code review with white space changes ignored as some extra indents were added due to a new `using`)

### CODE REVIEW

**Comment by danmosemsft** [comment]
These changes seem totally localized to HTTP/3 code - If that’s the case it might make sense to get into 5.0?

**Comment by scalablecory** [comment]
> These changes seem totally localized to HTTP/3 code - If that’s the case it might make sense to get into 5.0?

Yep, that's the intent.

File: src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs [hunk 1]
```diff
@@ -604,66 +601,77 @@ await using (stream.ConfigureAwait(false))
...
-                        return;
-                    default:
-                        await SkipUnknownPayloadAsync(frameType.GetValueOrDefault(), payloadLength).ConfigureAwait(false);
-                        break;
+                    (frameType, payloadLength) = await ReadFrameEnvelopeAsync().ConfigureAwait(false);
+
+                    switch (frameType)
+                    {
+                        case Http3FrameType.GoAway:
+                            await ProcessGoAwayFameAsync(payloadLength).ConfigureAwait(false);
```
Comment (danmosemsft) [inline src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs:644]: Typo..frame?

Comment (jkotalik) [reply src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs:644]: Nah Cory wants that fame 😄

**Review by jkotalik — APPROVED** [review]

**Review by geoffkizer — APPROVED** [review]

**Review by stephentoub — APPROVED** [review]

**Pull request merged**